docs/csharp/nullable-references.md

```diff
@@ -162,18 +164,27 @@ You must explicitly opt in to use these features in your projects. That provides
   - All new nullable warnings are enabled.
   - You can use the `?` suffix to declare a nullable reference type.
   - All other reference type variables are non-nullable reference types.
-  - The null forgiveness operator suppresses warnings for a possible assignment to `null`.
+  - The null forgiving operator suppresses warnings for a possible assignment to `null`.
 - *warnings*: The compiler performs all null analysis and emits warnings when code might dereference `null`.
   - All new nullable warnings are enabled.
   - Use of the `?` suffix to declare a nullable reference type produces a warning.
   - All reference type variables are allowed to be null. However, members have the *null-state* of *not-null* at the opening brace of all methods unless declared with the `?` suffix.
   - You can use the null forgiving operator, `!`.
-- *annotations*: The compiler doesn't perform null analysis or emits warnings when code might dereference `null`.
+- *annotations*: The compiler doesn't perform null analysis or emit warnings when code might dereference `null`.
   - All new nullable warnings are disabled.
   - You can use the `?` suffix to declare a nullable reference type.
   - All other reference type variables are non-nullable reference types.
   - You can use the null forgiving operator, `!`, but it has no effect.
 
+The nullable annotation context and nullable warning context can be set for a project using the [`<Nullable>` element](language-reference/compiler-options/language.md) in your *.csproj* file. This element configures how the compiler interprets the nullability of types and what warnings are generated. The following table shows the allowable values and summarizes the contexts they specify.
+
+| Context | Dereference warnings | Reference types | `?` suffix | `!` operator |
+| - | - | - | - | - |
+| `disabled` | Disabled | All are nullable | Can't be used | Has no effect |
+| `enabled` | Enabled | Non-nullable unless declared with `?` | Declares nullable type | Suppresses warnings for possible `null` assignment |
+| `warnings` | Enabled | All are nullable, but members are considered *not null* at opening brace of methods | Produces a warning |  Suppresses warnings for possible `null` assignment |
+| `annotations` | Disabled | Non-nullable unless declared with `?` | Declares nullable type | Has no effect |
```

Comment:
I tried adding a column for "Assignment warnings".  Does that help explain the warnings on annotations in that row?

```suggestion
| Context | Dereference warnings | Assignment warnings | Reference types | `?` suffix | `!` operator |
| - | - | - | - | - |
| `disabled` | Disabled | Disabled | All are nullable | Can't be used | Has no effect |
| `enabled` | Enabled | Enabled | Non-nullable unless declared with `?` | Declares nullable type | Suppresses warnings for possible `null` assignment |
| `warnings` | Enabled | Not applicable | All are nullable, but members are considered *not null* at opening brace of methods | Produces a warning |  Suppresses warnings for possible `null` assignment |
| `annotations` | Disabled | Disabled | Non-nullable unless declared with `?` | Declares nullable type | Has no effect |
```